tensorflow_lite_support/ios/task/vision/sources/TFLImageClassifier.m

```diff
@@ -0,0 +1,160 @@
+/* Copyright 2021 The TensorFlow Authors. All Rights Reserved.
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+ http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+ ==============================================================================*/
+#import "tensorflow_lite_support/ios/task/vision/sources/TFLImageClassifier.h"
+#import "tensorflow_lite_support/ios/sources/TFLCommon.h"
+#import "tensorflow_lite_support/ios/sources/TFLCommonUtils.h"
+#import "tensorflow_lite_support/ios/task/core/sources/TFLBaseOptions+Helpers.h"
+#import "tensorflow_lite_support/ios/task/processor/sources/TFLClassificationOptions+Helpers.h"
+#import "tensorflow_lite_support/ios/task/processor/utils/sources/TFLClassificationUtils.h"
+#import "tensorflow_lite_support/ios/task/vision/utils/sources/GMLImageUtils.h"
+
+#include "tensorflow_lite_support/c/task/vision/image_classifier.h"
+
+@interface TFLImageClassifier ()
+/** ImageClassifier backed by C API */
+@property(nonatomic) TfLiteImageClassifier *imageClassifier;
+@end
+
+@implementation TFLImageClassifierOptions
+@synthesize baseOptions;
+@synthesize classificationOptions;
+
+- (instancetype)init {
+  self = [super init];
+  if (self) {
+    self.baseOptions = [[TFLBaseOptions alloc] init];
```

Comment:
This initialization is better moved to TFLBaseOptions and TFLClassificationOptions, so that it can be shared with other tasks. Same for `c/task/vision/image_classifier.cc`, but you can do it in a separate CL for `c/task/vision/image_classifier.cc`.

Reply:
I have moved the initialisation to the respective classes.